pkg/informer/informer.go

```diff
@@ -453,21 +573,15 @@ func builtInInformableTypes() map[schema.GroupVersionResource]gvrPartialMetadata
 	}
 }
 
-func (d *DynamicDiscoverySharedInformerFactory) updateInformers() {
+func (d *genericDynamicDiscoverySharedInformerFactory[InformerType, ListerType, GenericInformerType]) updateInformers() {
 	klog.V(5).InfoS("Determining dynamic informer additions and removals")
 
 	latest := builtInInformableTypes()
 
 	// Get the unique set of Group(Version)Resources (version doesn't matter because we're expecting a wildcard
 	// partial metadata client, but we need a version in the request, so we need it here) and add them to latest.
-	crdGVRs := d.crdIndexer.ListIndexFuncValues(byGroupVersionResourceIndex)
-	for _, s := range crdGVRs {
-		parts := strings.Split(s, "/")
-		group := parts[0]
-		version := parts[1]
-		resource := parts[2]
-		gvr := gvrFor(group, version, resource)
-
+	crdGVRs := d.listGVRs()
+	for _, gvr := range crdGVRs {
```

Comment:
Having the interface know how to divulge GVRs and also then know how to turn them into partial metadata seems weird - just have one list call that knows how to return some super-type with both bits of information in there.

Reply:
I'd even return a `map[schema.GroupVersionresource]GVRPartialMetadata` then.